### PR TITLE
Modify PayID Format

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Migrate PayID format from:
`$xpring.money/georgewashington`
to
`georgewashington$xpring.money`

### Context of Change

Product requested change in format. 

Remove references to 'payment pointer' and rename a few classes / methods to reflect that we're not longer compliant with the payment pointer spec.

This is a breaking change, but PayID is experimental so I don't consider it a big deal.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

New spec implemented.

## Test Plan

New tests provided for CI. 

## Future Tasks
Uptake parsing logic and test cases in client libraries (JS / Swift / Java)

### CODE REVIEW

File: src/PayID/pay-id-components.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * A class which which encapsulates components of a Pay ID.
```
Comment (amiecorso) [inline src/PayID/pay-id-components.ts:2]: s/which which/which

File: test/PayID/pay-id-utils-test.ts [hunk 2]
```diff
@@ -1,66 +1,66 @@
...
 
 describe('PayIDUtils', function(): void {
-  it('parse payment pointer - host and path', function(): void {
-    // GIVEN a payment pointer with a host and a path.
-    const rawPaymentPointer = '$example.com/foo'
+  it('parse Pay ID - valid', function(): void {
+    // GIVEN a Pay ID with a host and a path.
+    const host = 'xpring.money'
+    const path = 'georgewashington'
+    const rawPayID = `${path}$${host}`
```
Comment (amiecorso) [inline test/PayID/pay-id-utils-test.ts:10]: is it just me or does host and path seem flipped here?

Comment (amiecorso) [reply test/PayID/pay-id-utils-test.ts:10]: oh nevermind, I see the whole point of this PR

File: src/PayID/pay-id-components.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ */
+export default class PayIDComponents {
+  /**
+   * Create a new PayIDComponents.
+   *
+   * @param host The host of the payment pointer, starting with a leading '/'.
```
Comment (hbergren) [inline src/PayID/pay-id-components.ts:8]: Does host start with a `leading '/'`, or does `path`?

File: test/PayID/pay-id-utils-test.ts [hunk 4]
```diff
@@ -1,66 +1,66 @@
...
-    assert.equal(paymentPointer?.path, '/foo')
+    assert.equal(payIDComponents?.host, host)
+    assert.equal(payIDComponents?.path, `/${path}`)
   })
 
-  it('parse payment pointer - well known path', function(): void {
-    // GIVEN a payment pointer with a well known path.
-    const rawPaymentPointer = '$example.com'
+  it('parse Pay ID - multiple dollar signs', function(): void {
+    // GIVEN a Pay ID with too many '$'.
```
Comment (hbergren) [inline test/PayID/pay-id-utils-test.ts:21]: I'm not sure this test is accurate. You _can_ have `$`s in the `path` of a URL. Maybe we should say the PayID spec disallows that, but I'm not sure it does right now.

Thoughts @austin-king ?

File: src/PayID/pay-id-components.ts (hunk 3 above)
Comment (keefertaylor) [reply src/PayID/pay-id-components.ts]: Done

**Comment by austin-king** [comment]
Can we hold off on implementation changes until we know how we're changing the format? Appreciate the speed but I don't want to waste your time if you have to reimplement some things.

For example, in this you parse user and host from this and treat the user as a path, we're not sure if we're going to stick with the payment pointer URL format behind the scenes yet

File: test/PayID/pay-id-utils-test.ts (hunk 4 above)
Comment (dino-rodriguez) [reply test/PayID/pay-id-utils-test.ts:21]: I think `$` should be a reserved char

Comment (austin-king) [reply test/PayID/pay-id-utils-test.ts:21]: Think it's best to disallow multiple $'s for PayID. I imagine this would be very confusing to any user.

**Comment by austin-king** [comment]
> @tedkalaw this PR looks good except we are only slicing off the last `/` to get the user.
> AKA for the URL https://wallet.com/payid/alice this PR would return:
> payid/alice$wallet.com
> but what we need is:
> alice$wallet.com/payid
> 
> Once we get that fixed we can merge this
> Idk what is best practice if Keefer is out today, should we just push to his branch?

@tedkalaw I read this PR wrong, the above comment is inaccurate. It provides the functionality as desired b/c it only concerns itself with mapping PayID -> URL not URL -> PayID. 

Approving

**Comment by tedkalaw** [comment]
i'm commandeering this ship

**Pull request merged**